internal/dag/policy.go

```diff
@@ -311,3 +311,41 @@ func prefixReplacementsAreValid(replacements []contour_api_v1.ReplacePrefix) (st
 
 	return "", nil
 }
+
+func rateLimitPolicy(in *contour_api_v1.RateLimitPolicy) *RateLimitPolicy {
+	if in == nil || in.Local == nil {
+		return nil
+	}
+
+	var rp RateLimitPolicy
+
+	var fillInterval time.Duration
+	switch in.Local.Unit {
+	case "second":
+		fillInterval = time.Second
+	case "minute":
+		fillInterval = time.Minute
+	case "hour":
+		fillInterval = time.Hour
+	default:
+		// TODO should not be possible due to kubebuilder validation
```

Comment:
maybe just debug log and return in the event this does get hit?

Reply:
Yeah, we could do that and return an empty rate limit policy, or return an error, record it as a warning in conditions, and not configure a rate limit policy. I guess the latter is probably best since it surfaces the issue to a user, just annoying to have to add an `error` return parameter to helpers like this :/

Reply:
I ended up returning an error and reporting it as an error condition on the proxy if this happens. Though unlikely, it probably still makes sense to have the check and the user-visible error.